components/datasource-selector/datasource-selector-map.service.ts

```diff
@@ -0,0 +1,181 @@
+import {Injectable} from '@angular/core';
+
+import VectorLayer from 'ol/layer/Vector';
+import {Feature} from 'ol';
+import {Fill, Stroke, Style} from 'ol/style';
+import {Vector} from 'ol/source';
+import {transform} from 'ol/proj';
+
+import {HsLogService} from '../../common/log/log.service';
+import {HsMapService} from '../map/map.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class HsDatasourcesMapService {
+  extentLayer: VectorLayer = new VectorLayer({
+    title: 'Datasources extents',
+    show_in_manager: false,
+    source: new Vector(),
+    style: function (feature, resolution) {
+      return [
+        new Style({
+          stroke: new Stroke({
+            color: '#005CB6',
+            width: feature.get('highlighted') ? 4 : 1,
+          }),
+          fill: new Fill({
+            color: 'rgba(0, 0, 255, 0.01)',
+          }),
+        }),
+      ];
+    },
+  });
+
+  /**
+   * @param HsMapService
+   * @param HsLogService
+   */
+  constructor(
+    private HsMapService: HsMapService,
+    private HsLogService: HsLogService
+  ) {
+    this.HsMapService.loaded().then((map) => this.init(map));
+  }
+
+  /**
+   * @param evt
+   */
+  mapPointerMoved(evt): void {
+    const features = this.extentLayer
+      .getSource()
+      .getFeaturesAtCoordinate(evt.coordinate);
+    let somethingChanged = false;
+    this.extentLayer
+      .getSource()
+      .getFeatures()
+      .forEach((feature) => {
+        if (feature.get('record').highlighted) {
+          feature.get('record').highlighted = false;
+          somethingChanged = true;
+        }
+      });
+    if (features.length) {
+      features.forEach((feature) => {
+        if (!feature.get('record').highlighted) {
+          feature.get('record').highlighted = true;
+          somethingChanged = true;
+        }
+      });
+    }
+    if (somethingChanged) {
```

Comment:
Is this call for setTimeout() actually useful?

Reply:
Hard to say.. Seems really useless to me. Will look at it.